File: problems/1790-check-if-one-string-swap-can-make-strings-equal/solution.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        if (s1.equals(s2)) {
            return true;
        }
        int first = -1, second = -1;
        int diffCount = 0;
        for (int i = 0; i < s1.length(); i++) {
            if (s1.charAt(i) != s2.charAt(i)) {
                diffCount++;
                if (first == -1) {
                    first = i;
                } else if (second == -1) {
                    second = i;
                }
            }
        }
        if (diffCount == 2 && s1.charAt(first) == s2.charAt(second) && s1.charAt(second) == s2.charAt(first)) {
            return true;
        }
        return false;
    }
}

File: problems/1790-check-if-one-string-swap-can-make-strings-equal/solution.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        int diff_count = 0;
        int first_diff = -1;
        int second_diff = -1;

        for (int i = 0; i < s1.length(); ++i) {
            if (s1[i] != s2[i]) {
                diff_count++;
                if (first_diff == -1) {
                    first_diff = i;
                } else if (second_diff == -1) {
                    second_diff = i;
                }
            }
        }

        if (diff_count == 0) {
            return true;
        } else if (diff_count == 2) {
            return (s1[first_diff] == s2[second_diff] && s1[second_diff] == s2[first_diff]);
        } else {
            return false;
        }
    }
};